RecoTauTag/HLTProducers/src/HLTPFDiJetCorrCheckerWithDiTau.cc

```diff
@@ -0,0 +1,132 @@
+/** Description: Check correlation between PFJet pairs and filtered PFTau pairs and store the PFJet pairs.
+For (j1, j2, t1, t2) where j1, j2 from the PFJet collection and t1, t2 from the filtered PFTau collection,
+the module checks if there is no overlap (within dRmin) between j1, j2, t1, t2, i.e. they are 4 different objects.
+In addition, the module imposes the following cuts:
+* mjjMin: the min invariant mass cut on (j1, j2)
+* extraTauPtCut: the leading tau pt cut on (t1, t2) (under the assumption t1, t2 are products of a subleading pt filter with minN = 2)
+The module stores j1, j2 of any (j1, j2, t1, t2) that satisfies the conditions above. */
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+#include "Math/GenVector/VectorUtil.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+#include "DataFormats/TauReco/interface/PFTau.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "CommonTools/Utils/interface/PtComparator.h"
+
+//
+// class definition
+//
+class HLTPFDiJetCorrCheckerWithDiTau : public edm::stream::EDProducer<> {
```

Comment:
Optional change: convert to `global`. In my understanding, for simple modules with const data members, `global` modules would be preferrable to `stream` modules (more info [here](https://github.com/cms-sw/cmssw/pull/33204#issuecomment-801266274)).

Reply:
OK